### PR TITLE
fix: check NEXT_DISABLE_NETLIFY_EDGE value rather than if truthy

### PR DESCRIPTION
**Pull request by charkour**
Hey Netlify Team, please let me know if you have comments or questions!

If you do not want to move forward with this change, what is the intended workaround for configuring Netlify Edge use based on context?

### Summary
This update now checks that `process.env.NEXT_DISABLE_NETLIFY_EDGE` is either `'true'` or `'false'` rather than checking that `process.env.NEXT_DISABLE_NETLIFY_EDGE` is truthy.

_Why?_

This allows Netlify Next.js users to have fine-grain control over setting environment variables for their deploys. By switching to the string value check, users of this runtime can disable Netlify Edge functions for a single branch, and leave it enabled for all others (or vice versa). This is especially helpful for debugging issues (e.g. Netlify Support #104849)

Fixes Netlify Support #106611

<details>
<summary>This is an example of how I would like to control the Netlify Edge based on context.</summary>

image

</details>

### Test plan

1. Deploy a site with `NEXT_DISABLE_NETLIFY_EDGE=false`. Inspect the build logs and look for a console.log in bright red. `To get the best performance and use Netlify Edge Functions`
2. Deploy a site with `NEXT_DISABLE_NETLIFY_EDGE=true`. No console.log will be present. 

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal

image

### Standard checks:

- [ ] Check the Deploy Preview's Demo site for your PR's functionality
- [x] Add docs when necessary

---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.

### CODE REVIEW

**Comment by charkour** [comment]
If this is approved, we should probably update the tests for this. I looked at the readme, and it doesn't seem to need any updates.

**Comment by charkour** [comment]
Hey @ascorbic,

Thanks for the quick reply. I see the logical bug in my suggested fix. Thank you for pointing that out.

Would you like me to make the update you suggested? I can see that this logic adds a bit of complexity to the code, but from a user perspective, I think this would be a nice change.

If you and your team do not want to make a change to this plugin, do you have a suggested workaround for unsetting NEXT_DISABLE_NETLIFY_EDGE based on context? I attempted to add an empty string in the Netlify UI to the env var but that doesn't work. Perhaps my team could add a pre-build script to unset the env var in our build pipeline. 

How would you like to move forward with this? Thanks!

**Comment by ascorbic** [comment]
I think it's a reasonable addition, so if you're up for making the change then we can do it in this PR

**Comment by charkour** [comment]
Great, sounds good. I'll update the logic in the files and will update the readme.

Are there specific tests you would like me to update? Please feel free to add changes yourself.

Thanks for the help!

**Comment by charkour** [comment]
Thanks for the help, @ascorbic.

File: packages/runtime/src/helpers/files.ts [hunk 1]
```diff
@@ -340,19 +340,19 @@ const baseServerReplacements: Array<[string, string]> = [
 const nextServerReplacements: Array<[string, string]> = [
   [
     `getMiddlewareManifest() {\n        if (this.minimalMode) return null;`,
-    `getMiddlewareManifest() {\n        if (this.minimalMode || (process.env.NEXT_DISABLE_NETLIFY_EDGE === 'false')) return null;`,
+    `getMiddlewareManifest() {\n        if (this.minimalMode || !(process.env.NEXT_DISABLE_NETLIFY_EDGE !== 'false' && process.env.NEXT_DISABLE_NETLIFY_EDGE !== '0')) return null;`,
```
Comment (ascorbic) [inline packages/runtime/src/helpers/files.ts:343]: You have the logic flipped on this

File: packages/runtime/src/helpers/files.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ export const matchesRewrite = (file: string, rewrites: Rewrites): boolean => {
 }
 
 export const getMiddleware = async (publish: string): Promise<Array<string>> => {
-  if (!process.env.NEXT_DISABLE_NETLIFY_EDGE) {
+  if (!(process.env.NEXT_DISABLE_NETLIFY_EDGE !== 'false' && process.env.NEXT_DISABLE_NETLIFY_EDGE !== '0')) {
```
Comment (orinokai) [inline packages/runtime/src/helpers/files.ts:63]:
```suggestion
  if (process.env.NEXT_DISABLE_NETLIFY_EDGE !== 'true' && process.env.NEXT_DISABLE_NETLIFY_EDGE !== '1') {
```

File: packages/runtime/src/helpers/files.ts [hunk 3]
```diff
@@ -346,19 +346,19 @@ const baseServerReplacements: Array<[string, string]> = [
 const nextServerReplacements: Array<[string, string]> = [
   [
     `getMiddlewareManifest() {\n        if (this.minimalMode) return null;`,
-    `getMiddlewareManifest() {\n        if (this.minimalMode || !process.env.NEXT_DISABLE_NETLIFY_EDGE) return null;`,
+    `getMiddlewareManifest() {\n        if (this.minimalMode || !(process.env.NEXT_DISABLE_NETLIFY_EDGE !== 'false' && process.env.NEXT_DISABLE_NETLIFY_EDGE !== '0')) return null;`,
```
Comment (orinokai) [inline packages/runtime/src/helpers/files.ts:349]:
```suggestion
    `getMiddlewareManifest() {\n        if (this.minimalMode || process.env.NEXT_DISABLE_NETLIFY_EDGE !== ‘true` && process.env.NEXT_DISABLE_NETLIFY_EDGE !== ‘1’) return null;`,
```

File: packages/runtime/src/helpers/files.ts [hunk 4]
```diff
@@ -346,19 +346,19 @@ const baseServerReplacements: Array<[string, string]> = [
...
   ],
   [
     `generateCatchAllMiddlewareRoute(devReady) {\n        if (this.minimalMode) return []`,
-    `generateCatchAllMiddlewareRoute(devReady) {\n        if (this.minimalMode || !process.env.NEXT_DISABLE_NETLIFY_EDGE) return [];`,
+    `generateCatchAllMiddlewareRoute(devReady) {\n        if (this.minimalMode || !(process.env.NEXT_DISABLE_NETLIFY_EDGE !== 'false' && process.env.NEXT_DISABLE_NETLIFY_EDGE !== '0')) return [];`,
```
Comment (orinokai) [inline packages/runtime/src/helpers/files.ts:353]:
```suggestion
    `generateCatchAllMiddlewareRoute(devReady) {\n        if (this.minimalMode || process.env.NEXT_DISABLE_NETLIFY_EDGE !== ‘true` && process.env.NEXT_DISABLE_NETLIFY_EDGE !== ‘1’) return [];`,
```

File: packages/runtime/src/helpers/files.ts [hunk 5]
```diff
@@ -346,19 +346,19 @@ const baseServerReplacements: Array<[string, string]> = [
...
   ],
   [
     `generateCatchAllMiddlewareRoute() {\n        if (this.minimalMode) return undefined;`,
-    `generateCatchAllMiddlewareRoute() {\n        if (this.minimalMode || !process.env.NEXT_DISABLE_NETLIFY_EDGE) return undefined;`,
+    `generateCatchAllMiddlewareRoute() {\n        if (this.minimalMode || !(process.env.NEXT_DISABLE_NETLIFY_EDGE !== 'false' && process.env.NEXT_DISABLE_NETLIFY_EDGE !== '0')) return undefined;`,
```
Comment (orinokai) [inline packages/runtime/src/helpers/files.ts:357]:
```suggestion
    `generateCatchAllMiddlewareRoute() {\n        if (this.minimalMode || process.env.NEXT_DISABLE_NETLIFY_EDGE !== ‘true` && process.env.NEXT_DISABLE_NETLIFY_EDGE !== ‘1’) return undefined;`,
```

File: packages/runtime/src/helpers/files.ts [hunk 6]
```diff
@@ -346,19 +346,19 @@ const baseServerReplacements: Array<[string, string]> = [
...
   ],
   [
     `getMiddlewareManifest() {\n        if (this.minimalMode) {`,
-    `getMiddlewareManifest() {\n        if (!this.minimalMode && process.env.NEXT_DISABLE_NETLIFY_EDGE) {`,
+    `getMiddlewareManifest() {\n        if (!this.minimalMode && (process.env.NEXT_DISABLE_NETLIFY_EDGE !== 'false' && process.env.NEXT_DISABLE_NETLIFY_EDGE !== '0')) {`,
```
Comment (orinokai) [inline packages/runtime/src/helpers/files.ts:361]:
```suggestion
    `getMiddlewareManifest() {\n        if (!this.minimalMode && (process.env.NEXT_DISABLE_NETLIFY_EDGE === 'true' || process.env.NEXT_DISABLE_NETLIFY_EDGE === '1')) {`,
```

File: packages/runtime/src/helpers/edge.ts [hunk 7]
```diff
@@ -214,7 +214,7 @@ export const writeEdgeFunctions = async (netlifyConfig: NetlifyConfig) => {
       path: '/_next/image*',
     })
   }
-  if (!process.env.NEXT_DISABLE_NETLIFY_EDGE) {
+  if (!(process.env.NEXT_DISABLE_NETLIFY_EDGE !== 'false' && process.env.NEXT_DISABLE_NETLIFY_EDGE !== '0')) {
```
Comment (orinokai) [inline packages/runtime/src/helpers/edge.ts:217]:
```suggestion
  if (process.env.NEXT_DISABLE_NETLIFY_EDGE !== ‘true` && process.env.NEXT_DISABLE_NETLIFY_EDGE !== ‘1’) {
```

**Review by orinokai — CHANGES_REQUESTED** [review]
apologies @charkour, the logic here is quite complicated and I think we got it wrong. i've had another look and have a few suggestions that keep things relatively neat and will account for undefined values, which our original suggestion didn't

**Comment by charkour** [comment]
Hey @orinokai,

Thank you for the reply! I'm just getting the notification from GitHub now saying you left a comment so apologies for my late response.

File: packages/runtime/src/helpers/edge.ts (hunk 7 above)
Comment (charkour) [reply packages/runtime/src/helpers/edge.ts:217]: I believe using DeMorgan's law, these two expressions should use the logical OR. I'll make that change.

File: packages/runtime/src/helpers/files.ts (hunk 2 above)
Comment (charkour) [reply packages/runtime/src/helpers/files.ts:63]: Same comment as before.

File: packages/runtime/src/helpers/edge.ts (hunk 7 above)
Comment (charkour) [reply packages/runtime/src/helpers/edge.ts:217]: Thanks! I'll make this change

**Comment by orinokai** [comment]
thanks @charkour, i'm just sorting out some failing tests this morning (looks like auth issues) and then hopefully we'll get this shipped

**Comment by charkour** [comment]
Sounds good. Thanks for your help!

**Review by orinokai — APPROVED** [review]

**Comment by charkour** [comment]
Thanks for getting this looked at @orinokai and @ascorbic!

**Pull request merged**